src/rpc/core_rpc_server.cpp

```diff
@@ -1316,9 +1518,19 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
-  bool core_rpc_server::on_get_info_json(const COMMAND_RPC_GET_INFO::request& req, COMMAND_RPC_GET_INFO::response& res, epee::json_rpc::error& error_resp)
+  bool core_rpc_server::on_get_info_json(const COMMAND_RPC_GET_INFO::request& req, COMMAND_RPC_GET_INFO::response& res, epee::json_rpc::error& error_resp, bool no_bootstrap)
   {
     PERF_TIMER(on_get_info_json);
+    bool r;
+    if (!no_bootstrap && use_bootstrap_daemon_if_necessary<COMMAND_RPC_GET_INFO>("JON_RPC", "get_info", req, res, r))
+    {
+      crypto::hash top_hash;
+      m_core.get_blockchain_top(res.height_without_bootstrap, top_hash);
+      ++res.height_without_bootstrap;
+      res.bootstrap_daemon_address = m_bootstrap_daemon_address;
```

Comment:
This could move above the `if` as it is also done in the "else" case

Reply:
No, `res` gets written in `use_bootstrap_daemon`, so it needs to be done after.

Reply:
> so it needs to be done after

Why? As far as I can tell, `use_bootstrap_daemon_if_necessary` only sets a few other unrelated fields in `res`, so it should be fine to set `res.bootstrap_daemon_address` _before_ calling it, no?

Reply:
No, it makes makes an RPC call to the bootstrap daemon which can set whatever values to `res`.

Reply:
Valid point 😳, I'm sorry.